rpc/connection.go

```diff
@@ -336,7 +336,12 @@ type ConnectionOpts struct {
 	// InitialReconnectBackoffWindow, if non zero, causes a random backoff
 	// before reconnecting. The random backoff timer is fast-forward-able by
 	// passing in a WithFireNow(ctx) into a RPC call.
-	InitialReconnectBackoffWindow time.Duration
+	InitialReconnectBackoffWindow func() time.Duration
+	// As the name suggests, we normally skip the "initial REconnect backoff"
+	// the very first time we try to connect. However, some callers instantiate
+	// new Connection objects after a disconnect, and they need the "first
```

Comment:
This is probably not mine to review, but IMO all reconnects should happen in this package, as opposed to making new `rpc.Connection` objects :-) Otherwise we might have redundant `rpc.Connection` lying around, and the redundant one might be trying to reconnect itself as well and continue with RPCs. We’ve had pretty stable reconnects for KBFS servers after moving the reconnect logic here.

Reply:
You'll have to talk to @mmaxim about the problems he ran into with reconnect.

I kind of get the impression that someone's going to want to rewrite this entire class, and then salt the earth so that the original never grows again :)

Reply:
> I kind of get the impression that someone's going to want to rewrite this entire class

Yeah I’ve had the impulse to do so a few times, but was afraid I was gonna introduce bugs or it would meet resistance so I ended up never trying 😅